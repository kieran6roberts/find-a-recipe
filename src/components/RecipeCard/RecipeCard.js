import React from "react";
import { FaPlus } from "react-icons/fa";
import classes from "./RecipeCard.module.css";

const RecipeCard = ({ id, title, image, save }) => {
  return (
    <li key={id}  className={classes.item} id={id}>
      <div className={classes.header}>
        <h2 className={classes.name}>{title}</h2>
        <button onClick={save} className={classes.btn}>
          <FaPlus className={classes.saveIcon} color="#333" />
        </button>
      </div>

      <span className={classes.divider} />
      
      <img src={image} alt={`${title}`} className={classes.image}></img>
    </li>
  )
};

export default RecipeCard;