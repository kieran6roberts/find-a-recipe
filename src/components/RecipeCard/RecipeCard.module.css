.item {
  display: flex;
  flex-direction: column;
}

.header,
.subHeader {
  display: flex;
  align-items: center;
}

.header {
  padding: var(--flow-lg) 0 0;
  justify-content: space-between;
}

.subHeader {
  justify-content: space-between;
  background-color: var(--light1);
}

.name {
  font-size: var(--text-md);
  font-weight: 700;
  text-transform: uppercase;
  width: 80%;
}

.serve {
  font-size: var(--text-sm);
  text-transform: capitalize;
  font-weight: 400;
}

.name:not:nth-of-type(1) {
  padding-top: var(--flow-lg);
}

.btn,
.btnList{
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0.5rem;
  border: none;
  transition: all 100ms ease-in;
  background-color: var(--primary-color);
}

.btn:before {
  content: "save";
  position: absolute;
  font-size: var(--text-xs);
  text-align: center;
  width: 2.5rem;
  top: 0;
  opacity: 0;
  transform: translateY(0rem);
  transition: all 100ms ease-in;
}

.btn:hover:before,
.btnList:hover:before {
  opacity: 1;
  transform: translateY(-1.5rem);
}

/* save btn was pressed */
.saved {
  background-color: #f00;
}

.divider, 
.dividerTitle {
  height: 1px;
  margin: var(--flow-sm) 0;
}

.dividerTitle {
  width: 100%;
  background-color: var(--primary-color);
}

.divider {
  width: 100%;
  background-color: var(--light2);
}

.flexWrapper {
  display: flex;
  align-items: flex-start;
}

.image {
  width: 40%;
  max-width: 250px;
  object-fit: cover;
}

.ingredientHeader {
  font-size: var(--text-xs);
  text-transform: uppercase;
  margin-right: 1rem;
}

.ingredientList,
.instructionList {
  display: none;
  visibility: hidden;
  opacity: 0;
  list-style: inside;
  text-transform: capitalize;
  font-size: var(--text-xs);
}
/* open list when button clicked*/
.open {
  display: block;
  visibility: visible;
  opacity: 1;
}

.instructionHeader {
  font-size: var(--text-xs);
  margin-right: 1rem;
  text-transform: uppercase;
}

.instruction {
  padding: var(--padding-input) 0;
}

