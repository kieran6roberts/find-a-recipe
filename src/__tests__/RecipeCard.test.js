import React from 'react';
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import RecipeCard from "../components/RecipeCard/RecipeCard";
import ButtonProvider from "../hooks/ButtonContext";

describe("<RecipeCard /> that outputs recipe information", () => {
  test("check empty prop returns no information text content", () => {
    render(
      <ButtonProvider>
        <RecipeCard />
      </ButtonProvider>
      );

    expect(screen.getByText("Ingredients not available for this recipe")).toBeInTheDocument();
    expect(screen.getByText("No instructions for this recipe")).toBeInTheDocument();
  });
  test("snapshot", () => {
    const tree = renderer.create(
      <ButtonProvider>
      <RecipeCard
      id="1"
      title="test title"
      image="img.test"
      save={jest.fn()}
      deleteIcon
      steps={[ {step: "test1", number: "test1"}, {step: "test2", number: "test2"} ]}
      ready="45 minutes"
      ingredients={[ {name: "test 1"}, {name: "test2"}]}
      ></RecipeCard>
    </ButtonProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("check document text based on props", () => {
      render(
        <ButtonProvider>
          <RecipeCard
          id="1"
          title="test title"
          image="img.test"
          save
          deleteIcon
          steps={[ {step: "test1", number: "test1"}, {step: "test2", number: "test2"} ]}
          ready="45 minutes"
          ingredients={[ {name: "test 1"}, {name: "test2"}]}
          ></RecipeCard>
        </ButtonProvider>);

      expect(screen.getByText(/test title/)).toBeInTheDocument();
      expect(screen.getByText("test1")).toBeInTheDocument();
      expect(screen.getByText("test2")).toBeInTheDocument();
      expect(screen.queryByText("No instructions for this recipe")).toBeNull();
      expect(screen.queryByText("Ingredients not available for this recipe")).not.toBeInTheDocument();
    });

    test("click save event", () => {
      const onClick = jest.fn();
      render(
        <ButtonProvider>
          <RecipeCard
            ingredients={[ {name: "test 1"}, {name: "test2"}]}
            steps={[ {step: "test1", number: "test1"}, {step: "test2", number: "test2"} ]}
            save={onClick}
            icon="save">
            </RecipeCard>
        </ButtonProvider>);

      const buttons = screen.queryAllByRole("button")
      fireEvent.click(buttons[0]);
      expect(onClick).toHaveBeenCalledTimes(1);

    });
});