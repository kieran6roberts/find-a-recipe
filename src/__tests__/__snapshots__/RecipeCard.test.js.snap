// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeCard /> that outputs recipe information snapshot 1`] = `
<li
  className="item"
  id="1"
>
  <div
    className="header"
  >
    <h2
      className="name"
    >
      test title
       
      <span
        className="serve"
      >
        (time to serve: 
        45 minutes
        )
      </span>
    </h2>
    <button
      className="btn"
      onClick={true}
    >
      <svg
        className="icon"
        color="fff"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "fff",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
        />
      </svg>
    </button>
  </div>
  <span
    className="dividerTitle"
  />
  <img
    alt="test title"
    className="image"
    src="img.test"
  />
  <span
    className="divider"
  />
  <div
    className="subHeader"
  >
    <h3
      className="ingredientHeader"
    >
      ingredients
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      <svg
        className="icon"
        color="fff"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "fff",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
        />
      </svg>
    </button>
  </div>
  <ul
    className="ingredientList 
                      false"
  >
    <li
      className="ingredients"
    >
      test 1
      , 
    </li>
    <li
      className="ingredients"
    >
      test2
      , 
    </li>
  </ul>
  <span
    className="divider"
  />
  <div
    className="subHeader"
  >
    <h3
      className="instructionHeader"
    >
      instructions
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      <svg
        className="icon"
        color="fff"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "fff",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
        />
      </svg>
    </button>
  </div>
  <ul
    className="instructionList 
                      false"
  >
    <li
      className="instruction"
    >
      test1
    </li>
    <li
      className="instruction"
    >
      test2
    </li>
  </ul>
</li>
`;
